### PR TITLE
[Keyboard] Add Moondrop Dash75

### PR DESCRIPTION
**Pull request by vinorodrigues**
## Description

Add the Moondrop Dash75 - an innovative 75% keyboard with integrated USB Hub and headphone DAC/Amp

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* *N/A*

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/moondrop/dash75/r1/config.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+/* disable debug print */
+//#define NO_DEBUG
+
+/* disable print */
+//#define NO_PRINT
+
+/* disable action features */
+//#define NO_ACTION_LAYER
+//#define NO_ACTION_TAPPING
+//#define NO_ACTION_ONESHOT
```
Comment (drashna) [inline keyboards/moondrop/dash75/r1/config.h:24]:
```suggestion
```

File: keyboards/moondrop/dash75/r1/config.h [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2023 moondroplab (@moondroplab)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define SERIAL_NUM "Dash 75"
```
Comment (zvecr) [inline keyboards/moondrop/dash75/r1/config.h:6]: SERIAL_NUM isnt a thing
```suggestion
```

Comment (zvecr) [inline keyboards/moondrop/dash75/r1/config.h:6]: SERIAL_NUM isnt a thing
```suggestion
```

File: keyboards/moondrop/dash75/keymaps/via/keymap.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        ],
+        [
+            "_______", "_______", "_______", "_______", "_______",            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",            "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",            "_______", "_______", "_______",
+            "_______", "_______", "_______",                       "_______",                                  "_______", "_______",            "_______", "_______", "_______"
+        ]
```
Comment (zvecr) [inline keyboards/moondrop/dash75/keymaps/via/keymap.json:30]: No point including layers that cannot be accessed
```suggestion
        ]
```

File: keyboards/moondrop/dash75/keymaps/default/keymap.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        ],
+        [
+            "_______", "_______", "_______", "_______", "_______",            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",            "_______", "_______",
+            "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______", "_______",            "_______", "_______", "_______",
+            "_______", "_______", "_______",                       "_______",                                  "_______", "_______",            "_______", "_______", "_______"
+        ]
```
Comment (zvecr) [inline keyboards/moondrop/dash75/keymaps/default/keymap.json:24]:
```suggestion
        ]
```

File: keyboards/moondrop/dash75/keymaps/via/config.h [hunk 5]
```diff
@@ -0,0 +1,7 @@
+// Copyright 2023 moondroplab (@moondroplab)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 2
```
Comment (zvecr) [inline keyboards/moondrop/dash75/keymaps/via/config.h:6]: Should go with default as 4 layers fit
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/moondrop/dash75/r1/config.h (hunk 2 above)
Comment (vinorodrigues) [reply keyboards/moondrop/dash75/r1/config.h:6]: sorry ... typo ... should be `SERIAL_NUMBER`.  it's used in tmk_core as part of the usb descriptor code.  in this brand it's used to distinguish from the embedded DAC
image
*(will populate the "serial Number" blank in the above picture)*

File: keyboards/moondrop/dash75/keymaps/via/keymap.json (hunk 3 above)
Comment (vinorodrigues) [reply keyboards/moondrop/dash75/keymaps/via/keymap.json:30]: * the blanks on the `via` keymap are intentional ... in VIA one can map keys post initial population of the dynamic eeprom allocation.  this allows for the user to add their own "Fn" functionality if they want.
* agree that it should go from the `default` keymap.

File: keyboards/moondrop/dash75/keymaps/via/config.h (hunk 5 above)
Comment (vinorodrigues) [reply keyboards/moondrop/dash75/keymaps/via/config.h:6]: agree

File: keyboards/moondrop/dash75/keymaps/default/keymap.json (hunk 4 above)
Comment (vinorodrigues) [reply keyboards/moondrop/dash75/keymaps/default/keymap.json:24]: agree

File: keyboards/moondrop/dash75/r1/config.h (hunk 2 above)
Comment (vinorodrigues) [reply keyboards/moondrop/dash75/r1/config.h]: *(as above / same item)*

File: keyboards/moondrop/dash75/keymaps/via/keymap.json (hunk 3 above)
Comment (zvecr) [reply keyboards/moondrop/dash75/keymaps/via/keymap.json:46]: You are incorrect in this being a requirement, it hasnt been for quite a while.

https://github.com/qmk/qmk_firmware/blob/master/quantum/dynamic_keymap.c#L158

This is why the layers can be removed, and why it is being requested.

Comment (vinorodrigues) [reply keyboards/moondrop/dash75/keymaps/via/keymap.json:46]: ahhh... right.  removing.

File: keyboards/moondrop/dash75/r1/config.h [hunk 6]
```diff
@@ -0,0 +1,10 @@
+// Copyright 2023 moondroplab (@moondroplab)
```
Comment (drashna) [inline keyboards/moondrop/dash75/r1/config.h:1]: is the r1 revision actually needed?   Eg, is there a planned r2, etc?

Comment (vinorodrigues) [reply keyboards/moondrop/dash75/r1/config.h:1]: yes - the next variant is in development

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**